### PR TITLE
Integration Fixes For Schema Responses

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

During initial integration testing I encountered three minor bugs. This PR fixes them.

## Changes

- `registration_schema` serialization was wrong and just outside of test ranges, simple fix
- If you PATCHed without a `revisions_response` dict it would 500 instead of no-oping
- We didn't add `SCHEMA_RESPONSE_READ/WRITE` to the scope config properly, so token auth didn't work. 
- `get_object` for the schema detail endpoint 500ed instread of 404ing for bad ids.
- We forgot to allow PATCHing for `revision_justifications`!

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify SchemaResponses are creatable and updatable via the API.
- Verify you can use the Bearer auth properly.

## Side Effects

None that I know of.

## Ticket

None.

### CODE REVIEW

File: api/schema_responses/serializers.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ class RegistrationSchemaResponseSerializer(JSONAPISerializer):
     date_modified = VersionedDateTimeField(source='modified', required=False)
     revision_justification = ser.CharField(required=False)
     updated_response_keys = ser.JSONField(required=False, read_only=True)
-    reviews_state = ser.ChoiceField(choices=['revision_in_progress', 'revision_pending_admin_approval', 'revision_pending_moderation', 'approved'], required=False)
+    reviews_state = ser.ChoiceField(choices=ApprovalStates.char_field_choices(), required=False)
```
Comment (Johnetordoff) [inline api/schema_responses/serializers.py:38]: Drive-by clean-up

File: osf/models/schema_response.py [hunk 2]
```diff
@@ -228,14 +227,13 @@ def update_responses(self, updated_responses):
         if updated_responses:
             raise ValueError(
                 'Encountered unexpected keys while trying to update responses for '
-                f'SchemaResponse with id [{self.id}]: {", ".join(updated_responses.keys())}'
```
Comment (Johnetordoff) [inline osf/models/schema_response.py:231]: Sending people the PK is not useful

Comment (jwalz) [reply osf/models/schema_response.py:231]: can you find and replace all instances of `self.id` in error messages with `self._id`?

Comment (jwalz) [reply osf/models/schema_response.py:231]: or else omit this change and I'll get it

**Review by jwalz — APPROVED** [review]
